src/utils/helpers/addFilterElement.ts

```diff
@@ -0,0 +1,38 @@
+import { IFilterApply, IRangeFilter } from 'interface/filterInterface'
+
+export const addFilterCheckbox = (prevState: IFilterApply | null, title: string, value: string) => {
+  if (!prevState) return { [title]: [value] }
+
+  const removeElement = Object.fromEntries(Object.entries(prevState).filter(element => element[0] !== title))
+  const preTitle = prevState[title as keyof IFilterApply] as string[] || []
+  const hasValue = preTitle?.find(filterValue => value === filterValue)
+  const removeValue = preTitle.filter(filterValue => value !== filterValue)
+
+  if (hasValue) {
+    if (preTitle.length > 1) {
+      return { ...prevState, [title]: [...removeValue] }
+    }
+
+    return { ...removeElement }
+  }
+
+  return { ...prevState, [title]: [...preTitle, value] }
+}
+
+export const addFilterRange = (prevState: IFilterApply | null, title: string, value: IRangeFilter) => {
+  if (!prevState) return { [title]: value }
+
+  const removeElement = Object.fromEntries(Object.entries(prevState).filter(element => element[0] !== title))
```

Comment:
I'm shure you don't need to use `Object.entries` inside `Object.fromEntries`. I believe exists better way

Reply:
fixed